src/main/java/redis/clients/jedis/ClientSideCache.java

```diff
@@ -10,12 +10,21 @@
 
 public class ClientSideCache {
 
-  private final Map<ByteBuffer, Object> cache = new HashMap<>();
+  private final Map<ByteBuffer, Object> cache;
 
-  protected ClientSideCache() {
+  public ClientSideCache() {
+    this.cache = new HashMap<>();
   }
 
-  protected void invalidateKeys(List list) {
+  /**
+   * For testing purpose only.
+   * @param map 
+   */
+  ClientSideCache(Map<ByteBuffer, Object> map) {
```

Comment:
WDYT about using an extends in the test tree instead, and keeping this out of public facing code?

Reply:
Will see.